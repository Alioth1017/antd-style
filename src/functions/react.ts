import _styled from '@emotion/styled';

import type { CreateStyled } from '@/types';

export const styled = _styled as CreateStyled;
export { css as reactCss, ThemeProvider as EmotionThemeProvider, withTheme } from '@emotion/react';
