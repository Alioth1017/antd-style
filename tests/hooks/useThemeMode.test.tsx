import { renderHook } from '@testing-library/react';
import { AppContainer, useThemeMode } from 'antd-style';

describe('useThemeMode', () => {
  it('没有包裹 AppContainer 会抛出错误', () => {
    expect(() => renderHook(useThemeMode)).toThrowErrorMatchingSnapshot();
  });

  it('包裹 AppContainer 后正常可以正常调用', () => {
    const { result } = renderHook(useThemeMode, { wrapper: AppContainer });
    expect(result.current.appearance).toEqual('light');
    expect(result.current.themeMode).toEqual('light');
  });

  it('AppContainer 传入值', () => {
    const { result } = renderHook(useThemeMode, {
      wrapper: ({ children }) => <AppContainer themeMode={'auto'}>{children}</AppContainer>,
    });
    expect(result.current.appearance).toEqual('light');
    expect(result.current.themeMode).toEqual('auto');
  });
});
