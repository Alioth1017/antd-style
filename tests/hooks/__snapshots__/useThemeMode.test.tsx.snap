// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useThemeMode 没有包裹 AppContainer 会抛出错误 1`] = `"[WrapperError]你可能没有使用 <AppContainer /> 组件，请确保在组件顶层包裹 AppContainer 组件后再使用该 hooks。"`;
